jax/interpreters/xla.py

```diff
@@ -954,12 +955,12 @@ def f(c, *xla_args, **params):
                                           stage_out=True)  # type: ignore
       xla_consts = _xla_consts(c, consts)
       outs = jaxpr_subcomp(c, jaxpr, None, axis_env, xla_consts, '', *xla_args)
-    if multiple_results:
+    if multiple_results or len(outs) > 1:
```

Comment:
This seems surprising. Don't we want `multiple_results` to indicate exactly when outputs should be tupled, and for it to be an error if multiple outputs are produced by the traced function when `multiple_results` is not indicated? The change on this line, together with deleting the assertion that was just below, seems like information is becoming more disorganized/complicated, e.g. in that `multiple_results` doesn't seem to mean what it did previously. I might just be missing something...

Reply:
Perhaps the confusion here is that this change does two things: (1) make xla translations that accept avals, and (2) allow `lower_fun` to be used for functions that return multi-buffered objects, in which case a tuple is required regardless of `multiple_results`.

Since the output avals are not available to inspect here, I don't know of any other way to support this than to assume that multiple buffers should be returned as a tuple. What do you think?

Reply:
Hm I see.

Actually, we might be able to inspect the output avals: `out_avals = [v.aval for v in jaxpr.outvars]`. Would that work?

Reply:
Good idea - done!